File: business/src/main/java/de/davelee/trams/business/request/AdjustDifficultyLevelRequest.java
package de.davelee.trams.business.request;

import lombok.*;

/**
 * This class is part of the TraMS Business REST API. It represents a request to adjust the difficulty level
 * for a particular company.
 * @author Dave Lee
 */
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class AdjustDifficultyLevelRequest {

    /**
     * The name of the company to adjust the difficulty level for.
     */
    private String company;

    /**
     * The new difficulty level which should be used for this company (can be EASY, MEDIUM or HARD).
     */
    private String difficultyLevel;

}

File: desktop/src/app/options/adjustDifficultyLevel.request.ts
/**
 * This class defines a model for requests to adjust the difficulty level in TraMS.
 */
export class AdjustDifficultyLevelRequest {

    private company: string;
    private difficultyLevel: string;

    /**
     * Construct a new request to adjust the difficulty level.
     * @param company the company that we should adjust the difficulty level for.
     * @param difficultyLevel the new difficulty level that should be set.
     */
    constructor(company: string, difficultyLevel: string) {
        this.company = company;
        this.difficultyLevel = difficultyLevel;
    }
}